src/metabase/models/setting.clj

```diff
@@ -467,28 +470,67 @@
 
 (def ^:private ^:dynamic *disable-cache* false)
 
+(def ^:private ^:dynamic *disable-init* false)
```

Comment:
My preference is to make `get` build on `read-setting`, rather than the other way around. This way we can just get rid of the dynamic var, which I agree is a big smell. I've updated the docstring that you pointed out and going to leave it at that for now.